### PR TITLE
hibernate-05

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

File: src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private UserService userService = new UserServiceImpl();
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        byte[] salt = HashUtil.getSalt();
+        String hashedPassword = HashUtil.hashPassword(password, salt);
```
Comment (mfedenko09) [inline src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java]:
```suggestion
        
```
You should create salt and hash password in UserService add() method.

File: src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        user.setPassword(hashedPassword);
+        user.setSalt(salt);
```
Comment (mfedenko09) [inline src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java]:
```suggestion
        user.setPassword(password);
```
same here

File: src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        userService.add(user);
+        return user;
+    }
+
+    @Override
+    public User login(String email, String password) {
+        User user = userService.getByEmail(email);
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        byte[] salt = user.getSalt();
+        String pass = user.getPassword();
```
Comment (mfedenko09) [inline src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java]:
```suggestion

```

File: src/main/java/com/hibernate/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import com.hibernate.service.UserService;
+import java.util.List;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao = new UserDaoImpl();
+
+    @Override
+    public User add(User user) {
```
Comment (mfedenko09) [inline src/main/java/com/hibernate/service/impl/UserServiceImpl.java:19]:
```suggestion
    public User add(User user) {
```
You should create salt and hash password in UserService add() method.

**Review by Fedich09 — CHANGES_REQUESTED** [review]

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/com/hibernate/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> getByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from User "
+                    + "where email = :email", User.class)
+                    .setParameter("email", email).uniqueResultOptional();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/hibernate/dao/impl/UserDaoImpl.java]:
```suggestion
            return session.createQuery("from User where email = :email", User.class)
                    .setParameter("email", email)
                    .uniqueResultOptional();
```

File: src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    public User register(String email, String password) {
+        User newUser = new User();
+        newUser.setEmail(email);
+        newUser.setPassword(password);
+        return userService.add(newUser);
+    }
+
+    @Override
+    public User login(String email, String password) {
+        User user = userService.getByEmail(email);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java]: Let's return optional from service too, and check here using isPresent()

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**